### PR TITLE
Use base path for urls on the client side

### PR DESCRIPTION
**Pull request by cabhishek**
- Use base path as per documented [here](https://github.com/ReactTraining/react-router/blob/d4aa590dd13848a857ca27508149552458c7cdfb/examples/pinterest/app.js#L7)
- ^^ option is preferred to [this](https://github.com/ReactTraining/react-router/issues/820)

### CODE REVIEW

File: packages/commuter-client/src/index.js [hunk 1]
```diff
@@ -12,11 +13,15 @@ import Notebook from "./Notebook";
 import Commuter from "./Commuter";
 import Discovery from "./Discovery";
 
+const history = useBasename(createHistory)({
+  basename: '/view'
```
Comment (cabhishek) [inline packages/commuter-client/src/index.js]: we can make this more configurable

Comment (rgbkrk) [inline packages/commuter-client/src/index.js]: This makes it so discovery is at /view/discovery now which I don't think is what we want.

However this is the right API for react router v3 for handling the base path as would be used with jupyterhub's /services/commuter endpoint.

Comment (cabhishek) [inline packages/commuter-client/src/index.js]: `/discovery` works with or without `/view`

Comment (rgbkrk) [inline packages/commuter-client/src/index.js]: What's at /view/discovery?

**Comment by rgbkrk** [comment]
Primary concerns here:

This makes _all_ routes live under `/view`, polluting the second level namespace. Pulling up `/view/discover` shows the discover page. We don't want that hanging out there and we certainly don't want any future routes to collide with the user namespace.

**Pull request closed without merging**